$NetBSD: patch-lib_roken_version-script.map,v 1.1 2024/09/18 14:50:12 hauke Exp $

Keep conditional libc fallback exports optional, retaining exact names.
Remove stale names with no libroken definition. Keep NetBSD 9 fix from
https://github.com/heimdal/heimdal/issues/106.

--- lib/roken/version-script.map.orig
+++ lib/roken/version-script.map
@@ -2,22 +2,20 @@
 	global:
 		arg_printusage;
 		arg_printusage_i18n;
-		cgetcap;
-		cgetclose;
-		cgetmatch;
-		cgetnum;
-		cgetset;
-		cgetustr;
+		[c]getcap;
+		[c]getclose;
+		[c]getmatch;
+		[c]getnum;
+		[c]getset;
+		[c]getustr;
 		ct_memcmp;
-		err;
-		errx;
+		[e]rr;
+		[e]rrx;
 		free_getarg_strings;
 		get_default_username;
 		get_window_size;
 		getarg;
 		getnameinfo_verified;
-		hex_decode;
-		hex_encode;
 		issuid;
 		k_getpwnam;
 		k_getpwuid;
@@ -33,19 +31,19 @@
 		print_time_table;
 		print_units_table;
 		rk_asnprintf;
-		rk_asprintf;
+		[r]k_asprintf;
 		rk_base64_decode;
 		rk_base64_encode;
-		rk_bswap16;
-		rk_bswap32;
-		rk_bswap64;
-		rk_cgetent;
-		rk_cgetstr;
+		[r]k_bswap16;
+		[r]k_bswap32;
+		[r]k_bswap64;
+		[r]k_cgetent;
+		[r]k_cgetstr;
 		rk_cloexec;
 		rk_cloexec_file;
                 rk_cloexec_dir;
                 rk_cloexec_socket;
-		rk_closefrom;
+		[r]k_closefrom;
 		rk_copyhostent;
 		rk_dns_free_data;
 		rk_dns_lookup;
@@ -60,41 +58,42 @@
 		rk_esetenv;
 		rk_estrdup;
 		rk_ewrite;
-		rk_flock;
-		rk_fnmatch;
+		[r]k_flock;
+		[r]k_fnmatch;
 		rk_free_environment;
-		rk_freeaddrinfo;
+		[r]k_freeaddrinfo;
 		rk_freehostent;
-		rk_freeifaddrs;
-		rk_gai_strerror;
-		rk_getaddrinfo;
-		rk_getifaddrs;
+		[r]k_freeifaddrs;
+		[r]k_gai_strerror;
+		[r]k_getaddrinfo;
+		[r]k_getifaddrs;
 		rk_getipnodebyaddr;
 		rk_getipnodebyname;
-		rk_getnameinfo;
-		rk_getprogname;
-		rk_glob;
-		rk_globfree;
+		[r]k_getnameinfo;
+		[r]k_getprogname;
+		rk_getpwnam_r;
+		[r]k_glob;
+		[r]k_globfree;
 		rk_hex_decode;
 		rk_hex_encode;
 		rk_hostent_find_fqdn;
-		rk_inet_ntop;
-		rk_inet_pton;
-		rk_localtime_r;
+		[r]k_inet_ntop;
+		[r]k_inet_pton;
+		[r]k_localtime_r;
 		rk_memset_s;
 		rk_mkdir;
-		rk_mkstemp;
+		[r]k_mkstemp;
 		rk_pid_file_delete;
 		rk_pid_file_write;
 		rk_pidfile;
 		rk_pipe_execv;
 		rk_random_init;
 		rk_read_environment;
-		rk_readv;
+		[r]k_readv;
 		rk_realloc;
-		rk_strerror;
-		rk_strerror_r;
-		rk_setprogname;
+		[r]k_strerror;
+		[r]k_strerror_r;
+		[r]k_setprogname;
 		rk_simple_execle;
 		rk_simple_execlp;
 		rk_simple_execve;
@@ -116,41 +115,40 @@
 		rk_socket_set_tos;
 		rk_socket_sockaddr_size;
 		rk_strcollect;
-		rk_strftime;
-		rk_strlcat;
-		rk_strlcpy;
+		[r]k_strftime;
+		[r]k_strlcat;
+		[r]k_strlcpy;
 		rk_strlwr;
-		rk_strndup;
-		rk_strnlen;
+		[r]k_strndup;
+		[r]k_strnlen;
 		rk_strpoolcollect;
 		rk_strpoolfree;
 		rk_strpoolprintf;
-		rk_strptime;
+		[r]k_strptime;
 		rk_strsep_copy;
-		rk_strsvis;
-		rk_strsvisx;
-		rk_strtoll;
-		rk_strtoull;
+		[r]k_strsvis;
+		[r]k_strsvisx;
+		[r]k_strtoll;
+		[r]k_strtoull;
 		rk_strunvis;
-		rk_strunvisx;
 		rk_strupr;
-		rk_strvis;
-		rk_strvisx;
-		rk_svis;
-		rk_timegm;
+		[r]k_strvis;
+		[r]k_strvisx;
+		[r]k_svis;
+		[r]k_timegm;
 		rk_timevaladd;
 		rk_timevalfix;
 		rk_timevalsub;
-		rk_tdelete;
-		rk_tfind;
-		rk_tsearch;
-		rk_twalk;
+		[r]k_tdelete;
+		[r]k_tfind;
+		[r]k_tsearch;
+		[r]k_twalk;
 		rk_undumpdata;
 		rk_unvis;
 		rk_vasnprintf;
-		rk_vasprintf;
-		rk_vis;
-		rk_vsnprintf;
+		[r]k_vasprintf;
+		[r]k_vis;
+		[r]k_vsnprintf;
 		rk_vstrcollect;
 		rk_wait_for_process;
 		rk_wait_for_process_timed;
@@ -184,7 +182,6 @@
 		rtbl_set_prefix;
 		rtbl_set_separator;
 		signal;
-		simple_execl;
 		tm2time;
 		unix_verify_user;
 		unparse_bytes;
@@ -194,13 +191,13 @@
 		unparse_time_approx;
 		unparse_units;
 		unparse_units_approx;
-		verr;
-		verrx;
-		vwarn;
-		vwarnx;
-		warn;
-		warnx;
-		writev;
+		[v]err;
+		[v]errx;
+		[v]warn;
+		[v]warnx;
+		[w]arn;
+		[w]arnx;
+		[w]ritev;
 	local:
 		*;
 };
